### ISSUE TITLE
Medtronic uploader

### ISSUE THREAD

**Issue by scottleibrand** [nightscout/android-uploader#77]

**Comment by PieterGit** [comment]
Any progress on this? Any way I can help with using nightscout with medtronic devices?

**Comment by jasoncalabrese** [comment]
See:
https://github.com/jberian/mmcommander
https://github.com/arbox0/MedtronicUploader
https://www.facebook.com/groups/cgminthecloud/permalink/693678477420078/?hc_location=ufi

Also some work to add support to the Nightscout Uploader using the carelink stick: https://github.com/nightscout/android-uploader/tree/minimed

**Comment by bewest** [comment]
It would be awesome to get [medtronic in python](https://github.com/bewest/decoding-carelink) ported to our sources;
https://github.com/nightscout/android-uploader/tree/minimed/core/src/main/java/com/nightscout/core/drivers/Medtronic

**Comment by dhermanns** [comment]
Sorry for bringing this up, Jason and Ben!
But is there another working solution beside mmcommander to upload medtronic data from medtronic enlight 2 transmitter to the nightscout backend? Or using a veo 554 pump?

My Texas Instruments Stick seems to make some problems and I would like to upload the data using the carelink stick as a backup. I think openaps should do this or Bens scripts, but I'm not sure about the details...

**Comment by scottleibrand** [comment]
Yes, the most common way of uploading data from a Medtronic pump in the context of closed looping is to build an OpenAPS rig and let it do it. The oref0-setup script has a MDT option that configures everything. If your pump is not capable of closed looping, the OpenAPS setup can still upload CGM and pump data.

**Comment by dhermanns** [comment]
Great. I will start with openaps then. Is there a documentation if you just wan't to configure openaps as an nightscout uploader for a medtronic pump?

Wouldn't it be great to have a docker container for the raspberry ready to run which does exactly the job above? I will try to build one if I succeed ;-)

**Comment by scottleibrand** [comment]
You can follow the normal OpenAPS docs, and set it up the same way you would a fully loopable pump.  It just won't enact any temps if the pump isn't in PC Connect mode.
